### PR TITLE
[docs][clang-tidy] Correct StrictMode example in modernize-use-std-print

### PR DESCRIPTION
**Pull request by MainakSil**
Updated the example in the `StrictMode` section of the clang-tidy check `modernize-use-std-print`.

The previous example incorrectly swapped the cast of signed and unsigned integers. Specifically:
- The signed integer `i` was being cast to `unsigned int`, and
- The unsigned integer `u` was being cast to `int`.

This correction ensures that the behavior of `std::print` with `StrictMode` enabled matches that of `printf`, by reversing the casts to maintain the correct signedness. 

Issue Refference
It solves #101397

### CODE REVIEW

File: clang-tools-extra/docs/clang-tidy/checks/modernize/use-std-print.rst [hunk 1]
```diff
@@ -109,7 +109,7 @@ Options
 
   .. code-block:: c++
 
-    std::print("{} {}\n", static_cast<unsigned int>(i), static_cast<int>(u));
+    std::print("{} {}\n", static_cast<int>(u), static_cast<unsigned int>(i));
```
Comment (nicovank) [inline clang-tools-extra/docs/clang-tidy/checks/modernize/use-std-print.rst:112]: The arguments shouldn't be swapped, I think the initial code block above must swap the `%d` and `%u` specifier for this example to make sense.

**Review by nicovank — COMMENTED** [review]
I think the example should be:
```cpp
printf("%u %d\n", i, u); // Change here.
// transforms to:
std::println("{} {}", static_cast<unsigned int>(i), static_cast<int>(u)); // No change here.
```

**Comment by nicovank** [comment]
cc @mikecrowe

**Comment by mikecrowe** [comment]
Thanks for fixing this @MainakSil. I wasn't aware of the bug report until now.

@nicovank is correct that only the format string needs to change as I got the format specifiers the wrong way round originally. Thanks for adding me to this change.

I copied the same mistake to `use-std-format.rst` if you'd like to fix that one too in this change. If not then I will do so.

Thanks.

**Review by mikecrowe — COMMENTED** [review]
LGTM. Thanks for fixing this.

**Review by nicovank — APPROVED** [review]

**Pull request merged**